server/src/main/java/org/elasticsearch/common/time/JavaDateFormatter.java

```diff
@@ -113,10 +113,14 @@ JavaDateFormatter getRoundupParser() {
     private List<DateTimeFormatter> createRoundUpParser(String format,
                                                         Consumer<DateTimeFormatterBuilder> roundupParserConsumer) {
         if (format.contains("||") == false) {
-            DateTimeFormatterBuilder builder = new DateTimeFormatterBuilder();
-            builder.append(this.parsers.get(0));
-            roundupParserConsumer.accept(builder);
-            return Arrays.asList(builder.toFormatter(locale()));
+            List<DateTimeFormatter> roundUpParsers = new ArrayList<>();
+            for (DateTimeFormatter parser : this.parsers) {
```

Comment:
How does this differ from the note above mentioning merges with `||` appending to the list of roundup formatters?

Reply:
so the comment mentions `In further merges (with ||) it will only append this one to a list.` When probably it should mention this in plural.

when a DateFormatter with multiple parsers like for instance `date_time` it will now also have a multiple round up parsers. When a `date_time` is then used in combination with other parsers, it will merge its parsers and round up parsers. 
for instance `date_time||date_time_no_millis`
`date_time` has 2 parsers and 2 round up parsers
`date_time_no_millis` also has 4 parsers and round up parsers
`date_time||date_time_no_millis` will merge to have 6 parsers and 6 round up parsers

I added a test case to confirm that all of them are being used.

The reason I initially only taken the first parser (` builder.append(this.parsers.get(0));` ) was that I tested for `JavaDateFormatter` created for a custom pattern. 
`DateFormatter.forPattern("YYYY")` where there no multiple parsers 

let me know if that makes it clear?